domain/cloudimagemetadata/state/state.go

```diff
@@ -0,0 +1,344 @@
+
+	findMetadataQuery := fmt.Sprintf(`
+WITH arch(id, architecture_name) AS (
+	SELECT id, name FROM architecture
+)
+SELECT &outputMetadata.* 
+FROM cloud_image_metadata
+JOIN arch ON cloud_image_metadata.architecture_id = arch.id
+WHERE %s;`, strings.Join(clauses, ` AND `))
+
+	findMetadataStmt, err := sqlair.Prepare(findMetadataQuery, outputMetadata{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var metadatas []outputMetadata
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, findMetadataStmt).GetAll(&metadatas)
+		if dberrors.IsErrNotFound(err) {
+			return cloudmetadataerrors.NotFound
+		}
+		return err
+	})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	result := make([]cloudimagemetadata.Metadata, len(metadatas))
+	for i, m := range metadatas {
+		result[i] = cloudimagemetadata.Metadata{
+			MetadataAttributes: cloudimagemetadata.MetadataAttributes{
+				Stream:          m.Stream,
+				Region:          m.Region,
+				Version:         m.Version,
+				Arch:            m.ArchitectureName,
+				VirtType:        m.VirtType,
+				RootStorageType: m.RootStorageType,
+				RootStorageSize: m.RootStorageSize,
+				Source:          m.Source,
+			},
+			Priority:     m.Priority,
+			ImageId:      m.ImageID,
+			CreationTime: m.CreatedAt,
+		}
+	}
+	return result, nil
+}
+
+// AllCloudImageMetadata retrieves all cloud image metadata from the database without applying any filter criteria.
+// Returns a slice of [cloudimagemetadata.Metadata] and any error encountered during the retrieval process.
+func (s *State) AllCloudImageMetadata(ctx context.Context) ([]cloudimagemetadata.Metadata, error) {
+	result, err := s.FindMetadata(ctx, cloudimagemetadata.MetadataFilter{})
+	if errors.Is(err, cloudmetadataerrors.NotFound) {
+		return nil, nil
+	}
+	return result, errors.Trace(err)
+}
+
+// tryCleanUpExpiredMetadata removes metadata records from the cloud_image_metadata table that are older than [ExpirationDelay].
+//
+// Custom images doesn't expire.
+func (s *State) tryCleanUpExpiredMetadata(ctx context.Context) error {
+	db, err := s.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	deleteQuery := fmt.Sprintf(`
+DELETE FROM cloud_image_metadata 
+WHERE source != '%s' AND  
+created_at < '%s';`, CustomSource, time.Now().Add(-ExpirationDelay).Format(time.DateTime))
+	deleteMetadataStmt, err := sqlair.Prepare(deleteQuery)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return errors.Trace(db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, deleteMetadataStmt).Run()
+		return err
+	}))
+}
+
+// ensuresArchitecturesLoaded initializes architecture data by loading it from the database if not already done.
+// architecture are readonly, inserted at the creation of the DB, so it should be ok to load them once at the start
+// of the controller.
+func (s *State) ensuresArchitecturesLoaded(ctx context.Context) (err error) {
```

Comment:
You don't need to do this. Just hardcode the values and then have a test that validates the integers are correct at test time. We've done this in the past, it keeps everything cleaner in the state.